### PR TITLE
test: port DTMF test to karma

### PR DESCRIPTION
**Pull request by fippo**
ports the DTMF test to karma and removes the legacy DTMF test (which was also a bit getSenders-ish)

I pledge to re-add a better test for the removed one ;-)

### CODE REVIEW

**Comment by fippo** [comment]
@henbos part of the tests i promised earlier.
@jan-ivar you seem to have a DTMF sender on video rtpsender in FF54 which is odd.

**Comment by fippo** [comment]
@KaptenJansson can you take a look at the first two commits? I noticed there was no linter for test/e2e/ and of course there were linting errors :-/

**Comment by fippo** [comment]
looks like the DTMF test is not as ready as I thought... yay for travis checking the stuff I don't check locally :-)
lint stuff in #603

File: test/e2e/dtmf.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    let pc1;
+    let pc2;
+
+    beforeEach(() => {
+      pc1 = new RTCPeerConnection(null);
+      pc2 = new RTCPeerConnection(null);
+
+      pc1.onicecandidate = e => pc2.addIceCandidate(e.candidate);
+      pc2.onicecandidate = e => pc1.addIceCandidate(e.candidate);
+      pc1.onnegotiationneeded = e => pc1.createOffer()
```
Comment (KaptenJansson) [inline test/e2e/dtmf.js:53]: Love es6 arrow functions together with promises :)

File: test/e2e/dtmf.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+      .then(stream => {
+        pc.addStream(stream);
+        const senders = pc.getSenders();
+        const dtmf = senders[0].dtmf;
+        expect(dtmf).not.to.equal(null);
+        expect(dtmf).to.have.property('insertDTMF');
+      });
+    });
+
+    /* Broken in Firefox 54 */
```
Comment (KaptenJansson) [inline test/e2e/dtmf.js:28]: So the strategy has now changed to disable the test entirely rather per browser?

**Review by KaptenJansson — APPROVED** [review]
LGTM with a question :)

File: test/e2e/dtmf.js (hunk 1 above)
Comment (fippo) [reply test/e2e/dtmf.js:53]: I just copied from @jan-ivar

File: test/e2e/dtmf.js (hunk 2 above)
Comment (fippo) [reply test/e2e/dtmf.js:28]: we still need to figure out a good pattern for disabling a test per browser :-/

Comment (fippo) [reply test/e2e/dtmf.js:28]: I found a nice way. Essentially files listing all expectations (similar to chrome layouttest) and whether they are passing combined with a custom karma reporter... will do in a followup PR

Comment (fippo) [reply test/e2e/dtmf.js:28]: https://github.com/fippo/karma-stability-reporter -- i don't think this deserves a npm name just yet

Comment (KaptenJansson) [reply test/e2e/dtmf.js:28]: Neat!

**Pull request merged**